sdk/nodejs/tests/runtime/install-package-tests.ts

```diff
@@ -0,0 +1,235 @@
+// Copyright 2024-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { randomInt } from "crypto";
+import execa from "execa";
+import * as fs from "fs/promises";
+import * as path from "path";
+import * as process from "process";
+import * as tmp from "tmp";
+import { pack } from "./pack";
+
+// Write a package.json that installs the local pulumi package and optional dependencies.
+async function writePackageJSON(
+    dir: string,
+    pulumiPackagePath: string,
+    dependencies: Record<string, string | undefined>,
+) {
+    const packageJSON = {
+        name: "install-package-tests",
+        version: "1.0.0",
+        license: "Apache-2.0",
+        dependencies: {
+            "@pulumi/pulumi": pulumiPackagePath,
+            ...dependencies,
+        },
+    };
+
+    await fs.writeFile(path.join(dir, "package.json"), JSON.stringify(packageJSON, undefined, 4));
+}
+
+async function writeTSConfig(dir: string) {
+    const tsconfigJSON = {
+        compilerOptions: {
+            strict: true,
+            target: "es2016",
+            module: "commonjs",
```

Comment:
Might be interesting to have an option here in the future for `node16` here to emit ESM